uaclient/entitlements/anbox.py

```diff
@@ -0,0 +1,80 @@
+from typing import Tuple
+
+from uaclient import event_logger, messages, system
+from uaclient.entitlements.repo import RepoEntitlement
+from uaclient.files.state_files import (
+    AnboxCloudCredentials,
+    AnboxCloudData,
+    anbox_cloud_credentials_file,
+)
+from uaclient.types import MessagingOperationsDict, StaticAffordance
+
+event = event_logger.get_event_logger()
+
+
+class AnboxEntitlement(RepoEntitlement):
+    name = "anbox-cloud"
+    title = "Anbox Cloud"
+    description = "Scalable Android in the cloud"
+    help_doc_url = "https://anbox-cloud.io"
+    repo_key_file = "ubuntu-pro-anbox-cloud.gpg"
+    repo_url_tmpl = "{}"
+    affordance_check_series = True
+    supports_access_only = True
+
+    @property
+    def static_affordances(self) -> Tuple[StaticAffordance, ...]:
+        return (
+            (
+                messages.SERVICE_ERROR_INSTALL_ON_CONTAINER.format(
+                    title=self.title
+                ),
+                lambda: system.is_container(),
+                False,
+            ),
+        )
+
+    @property
+    def messaging(self) -> MessagingOperationsDict:
+        if not self.access_only:
+            return {"post_enable": [messages.ANBOX_RUN_INIT_CMD.msg]}
+        else:
+            return {}
+
+    def _perform_enable(self, silent: bool = False) -> bool:
+        ret = super()._perform_enable(silent=silent)
+
+        if not ret:
+            return ret
+
+        directives = self.entitlement_cfg.get("entitlement", {}).get(
+            "directives", {}
+        )
+        anbox_images_directives = (
+            self.cfg.machine_token_file.entitlements.get("anbox-images", {})
+            .get("entitlement", {})
+            .get("directives", {})
+        )
+        anbox_images_resource_token = ""
+        for resource_token in self.cfg.machine_token_file.resource_tokens:
+            if resource_token.get("type") == "anbox-images":
+                anbox_images_resource_token = resource_token.get("token", "")
+                break
```

Comment:
This only works because anbox-images has `enableByDefault: true`, but I think we'd like to change that. Instead this can call `contract_client.get_resource_machine_access` to fetch the resource token from the server